File: Niveau 3/Niveau 3.8/encadre.py
#-------------------------------------------------------------------------------
# Name:        Nombre encadré
# Purpose:     France-IOI niveau 3.8
#
# Author:      Sébastien
#
# Created:     01/10/2022
#-------------------------------------------------------------------------------

def ecrire_crochets(n, e):
    if e == 0:
        print(n, end='')
    else:
        print('[', end='')
        ecrire_crochets(n, e-1)
        print(']', end='')

def main():
    n, e = map(int, input().split())
    ecrire_crochets(n, e)

if __name__ == '__main__':
    main()

File: Niveau 3/Niveau 3.8/encadre.hs
--------------------------------------------------------------------------------
-- Name:        Hydroélectricité
-- Purpose:     France-IOI niveau 3.8
--
-- Author:      Sébastien
--
-- Created:     01/10/2022
--------------------------------------------------------------------------------

import Control.Monad
import Data.List

main :: IO()
main = do  
    n:e:_ <- fmap (map read . words) getLine :: IO [Int]
    ecrire_crochets n e
    
ecrire_crochets :: Int -> Int -> IO()
ecrire_crochets n 0 = putStr $ show n
ecrire_crochets n e = do
       putStr "["
       ecrire_crochets n (e-1)
       putStr "]"

